app/views/handlers/list_collector_content.py

```diff
@@ -0,0 +1,20 @@
+from typing import Any
+
+from app.views.handlers.list_collector import ListCollector
+from app.views.handlers.question import Question
+
+
+class ListCollectorContent(ListCollector):
+    def _get_additional_view_context(self) -> Any:
+        return self.rendered_block["content"]
+
+    def handle_post(self) -> None:
+        if self._is_list_collector_complete():
+            self._routing_path = self.router.routing_path(
+                section_id=self._current_location.section_id,
+                list_item_id=self._current_location.list_item_id,
+            )
+            return super(Question, self).handle_post()
+        self.questionnaire_store_updater.add_completed_location()
+        self.questionnaire_store_updater.update_progress_for_dependent_sections()
+        self.questionnaire_store_updater.save()
```

Comment:
To fix x, this will need 
```
        if self._is_list_collector_complete():
            self._routing_path = self.router.routing_path(
                section_id=self._current_location.section_id,
                list_item_id=self._current_location.list_item_id,
            )
            super(Question, self).handle_post()
```

then in list repeating question you want to do the same logic as the list add question:
Create a helper method and reuse it in add and repeating question block.
```
        # Clear the answer from the confirmation question on the list collector question
        answer_ids_to_remove = self._schema.get_answer_ids_for_block(
            # Type ignore: for parent_location block_id is not none
            self.parent_location.block_id  # type: ignore
        )
        self.questionnaire_store_updater.remove_answers(answer_ids_to_remove)
        self.questionnaire_store_updater.remove_completed_location(self.parent_location)
```